lib/Sema/TypeCheckType.cpp

```diff
@@ -3804,39 +3804,7 @@ class UnsupportedProtocolVisitor
   }
 
   void visitIdentTypeRepr(IdentTypeRepr *T) {
-    if (T->isInvalid())
```

Comment:
I think you can remove UnsupportedProtocolVisitor and checkUnsupportedProtocolType() entirely

Reply:
Done :)